synapse/rest/media/v1/media_repository.py

```diff
@@ -355,14 +374,15 @@ def generate_local_exact_thumbnail(self, media_id, t_width, t_height,
         input_path = self.filepaths.local_media_filepath(media_id)
 
         thumbnailer = Thumbnailer(input_path)
-        t_byte_source = yield preserve_context_over_fn(
-            threads.deferToThread,
+        t_byte_source = yield make_deferred_yieldable(threads.deferToThread(
             self._generate_thumbnail,
             thumbnailer, t_width, t_height, t_method, t_type
-        )
+        ))
 
         if t_byte_source:
-            output_path = yield self.write_to_file(
+            t_width, t_height = t_byte_source.dimensions
```

Comment:
I'm confused.
* `t_byte_source` is the result from `_generate_thumbnail`
* `_generate_thumbnail` doesn't document its return type (grr) but apparently it's the result from `thumbnailer.crop` or `thumbnailer.scale`
* `crop` and `scale` return an ordinary BytesIO (at least according to their docs)
* BytesIO doesn't have a `dimensions` attribute

either something's lying, or you (still) haven't tested this. Or I'm being a total crank.

Reply:
why is this change happening, anyway?

Reply:
Yes, this was a spurious change, sowwy.

In other news, I've been looking at getting sytest to test both dynamic and static config, and it turns out that this dynamic code path doesn't work at all on master....

I think we should look into fixing/removing this functionality, but that should probably wait for a separate PR.

Reply:
> In other news, I've been looking at getting sytest to test both dynamic and static config, and it turns out that this dynamic code path doesn't work at all on master....

After massaging sytest a bit I've exercised that code path in a way that returns 200, so it seems no more broken than before, at least.